### DISCUSSION TITLE
OAuth login with desktop application

### DISCUSSION THREAD

**Discussion by Wardstein** [directus/directus#5139] [Archive]
Hi guys,

I want to use Directus (and our Gitlab) in a C# desktop application. First I set up Directus to be able to do a oAuth with our self-hosted gitlab, which for the website works perfectly fine.

Now I want to access gitlab and Directus from the C# application. For gitlab this is no problem, I do a proper oAuth authentication and get the two tokens. Then I thought I could use these tokens also for Directus, but this does not work (as Directus does not know anything about these tokens)...

When I call the /auth/oauth/gitlab endpoint in the browser I get the json output and access token (as I am logged in with the browser also in gitlab). But when I call this endpoint from within my C# application I am refered to the gitlab login page (as expected).
When I pass the gitlab token I have to the /auth/oauth/gitlab endoint, I get an invalid credentials error from Directus, as it thinks this is its own token.

Then I tried some reading in the auth endpoints code, and saw the redirect query parameter for the auth/oauth/<provider> endpoint. But this does not deliver any token information of the returned json to the redirected site.

I tested service a webpage from appication which calls the /auth/oauth/gitlab endpoint via ajax and returns the json data to my application, but then I would need to set the Access-Control-Allow-Origin headers on the Directus and Gitlab servers to make this work.

Is there any way how I could login to Directus via oAuth in my C# application?

Cheers,
Jonas

**Comment by marekhubeny** [comment]
Hi, please, did you solve it somehow? I have the same issue :(
